cookbooks/postgresql/attributes/default.rb

```diff
@@ -0,0 +1,281 @@
+    default['postgresql']['version'] = '9.2'
+    default['postgresql']['client']['packages'] = ['postgresql92', 'postgresql92-devel']
+    default['postgresql']['server']['packages'] = ['postgresql92-server']
+    default['postgresql']['contrib']['packages'] = ['postgresql92-contrib']
+  end
+
+  default['postgresql']['server']['service_name'] = "postgresql"
+
+when "suse"
+  if node['platform_version'].to_f <= 11.1
+    default['postgresql']['version'] = "8.3"
+    default['postgresql']['client']['packages'] = ['postgresql', 'rubygem-pg']
+    default['postgresql']['server']['packages'] = ['postgresql-server']
+    default['postgresql']['contrib']['packages'] = ['postgresql-contrib']
+  else
+    default['postgresql']['version'] = "9.1"
+    default['postgresql']['client']['packages'] = ['postgresql91', 'rubygem-pg']
+    default['postgresql']['server']['packages'] = ['postgresql91-server']
+    default['postgresql']['contrib']['packages'] = ['postgresql91-contrib']
+  end
+
+  default['postgresql']['dir'] = "/var/lib/pgsql/data"
+  default['postgresql']['server']['service_name'] = "postgresql"
+
+else
+  default['postgresql']['version'] = "8.4"
+  default['postgresql']['dir'] = "/etc/postgresql/8.4/main"
+  default['postgresql']['client']['packages'] = ["postgresql"]
+  default['postgresql']['server']['packages'] = ["postgresql"]
+  default['postgresql']['contrib']['packages'] = ["postgresql"]
+  default['postgresql']['server']['service_name'] = "postgresql"
+end
+
+case node['platform_family']
+when 'debian'
+  default['postgresql']['config']['listen_addresses'] = 'localhost'
+  default['postgresql']['config']['port'] = 5432
+  default['postgresql']['config']['max_connections'] = 100
+  default['postgresql']['config']['shared_buffers'] = '24MB'
+  default['postgresql']['config']['log_line_prefix'] = '%t '
+  default['postgresql']['config']['datestyle'] = 'iso, mdy'
+  default['postgresql']['config']['default_text_search_config'] = 'pg_catalog.english'
+  default['postgresql']['config']['ssl'] = true
+when 'rhel', 'fedora', 'suse'
+  default['postgresql']['config']['listen_addresses'] = 'localhost'
+  default['postgresql']['config']['port'] = 5432
+  default['postgresql']['config']['max_connections'] = 100
+  default['postgresql']['config']['shared_buffers'] = '32MB'
+  default['postgresql']['config']['logging_collector'] = true
+  default['postgresql']['config']['log_directory'] = 'pg_log'
+  default['postgresql']['config']['log_filename'] = 'postgresql-%a.log'
+  default['postgresql']['config']['log_truncate_on_rotation'] = true
+  default['postgresql']['config']['log_rotation_age'] = '1d'
+  default['postgresql']['config']['log_rotation_size'] = 0
+  default['postgresql']['config']['datestyle'] = 'iso, mdy'
+  default['postgresql']['config']['lc_messages'] = 'en_US.UTF-8'
+  default['postgresql']['config']['lc_monetary'] = 'en_US.UTF-8'
+  default['postgresql']['config']['lc_numeric'] = 'en_US.UTF-8'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.